src/backend/vulkan/src/lib.rs

```diff
@@ -895,7 +910,23 @@ impl adapter::PhysicalDevice<Backend> for PhysicalDevice {
                 || self.properties.device_id & info::intel::DEVICE_SKY_LAKE_MASK
                     == info::intel::DEVICE_SKY_LAKE_MASK);
 
-        let features = unsafe { self.instance.0.get_physical_device_features(self.handle) };
+        let mut descriptor_indexing_features = None;
+        let features = if let Some(ref get_device_properties) = self.instance.2 {
+            let features = vk::PhysicalDeviceFeatures::builder().build();
+            let mut features2 = vk::PhysicalDeviceFeatures2KHR::builder().features(features).build();
+            if self.supports_extension(*EXT_DESCRIPTOR_INDEXING) {
+                descriptor_indexing_features = Some(vk::PhysicalDeviceDescriptorIndexingFeaturesEXT::builder().build());
+                if let Some(ref mut mut_ref) = descriptor_indexing_features {
```

Comment:
since we know this to be `Some` at this point, we should not do `if`. Instead, let's do `descriptor_indexing_features.as_mut().unwrap()` or similar

Reply:
I thought that would be rejected by the borrow checker, but now I think about it, idk why...